### PR TITLE
Tutorial - LOJ 1089 (en)

### PR DESCRIPTION
**Pull request by Talha-Taki002**
#449

### CODE REVIEW

File: 1089/en.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+Note: **AA** is a valid segment.
+
+## Solution
+First let's simplify the problem a bit. Suppose there is no such segment like: **AA** 
+
+Let's think of each segment **AB** as of two points where **A** is the opening of the segment and **B** the ending of it. 
+
+Consider the set **S** of all opening and ending points of the segments sorted in increasing order. Iterating over the set from the begining, whenever we find an opening point **S[i]**, we increment the **counter** (starting with 0) by 1, indicating for any point **p** such that **S[i] <= p <= S[i+1]**,  if **S[i+1]** is an ending point, otherwise **S[i] <= p < S[i+1]** will be contained by that many of segments. If we find an ending point **S[i]**, we know for sure we had the opening point of that segment before, so it's time to close the segment, so we decrement the **counter** by 1, now indicating for any point **p** such that **S[i] < p <= S[i+1]**,  if **S[i+1]** is an ending point, otherwise **S[i] < p < S[i+1]** will be contained by that many of segments.
```
Comment (rebornplusplus) [inline 1089/en.md:13]: Let's take a step back and see if we can simplify this paragraph. I understand what you are trying to explain here, but I feel there are too many complicated sentences jumbled together. Let's explain it simply.
While doing so, let's also deviate away from any implementation details, for example, not bother explaining how you may write this in code.

File: 1089/en.md [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+We can sort our queries so that we can calculate answers for them in increasing order as we iterate over the set **S**.
+
+That was the general approach. Now, solving the original problems is all about handling the exception case where we may have the same point for many openings and endings. See the code and documentation for understading how to handle duplicate points.
```
Comment (rebornplusplus) [inline 1089/en.md:17]: ~documentation~

comments

**Review by rebornplusplus — CHANGES_REQUESTED** [review]
It looks good overall. Does the code pass in the website?

File: 1089/en.md [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+## Complexity
+- Time Complexity: O(T * (N $log{_2}{N}$ + Q $log{_2}{Q}$).
```
[the remark this answers is not in the document]
Comment (Talha-Taki002) [reply 1089/en.md:20]: How would you suggest to write down the complexity?

File: 1089/en.md (hunk 2 above)
Comment (Talha-Taki002) [reply 1089/en.md:17]: Fixed!

File: 1089/en.md (hunk 1 above)
Comment (Talha-Taki002) [reply 1089/en.md:13]: Simplified!

File: 1089/en.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (rebornplusplus) [reply 1089/en.md:24]: Just write it in plain English. Use `lg` instead of `log_2` if you prefer it over `log`. In CS, `log` typically translates to `log_2`.

File: 1089/en.md [hunk 4]
```diff
@@ -10,15 +10,19 @@ First let's simplify the problem a bit. Suppose there is no such segment like: *
 
 Let's think of each segment **AB** as of two points where **A** is the opening of the segment and **B** the ending of it. 
 
-Consider the set **S** of all opening and ending points of the segments sorted in increasing order. Iterating over the set from the begining, whenever we find an opening point **S[i]**, we increment the **counter** (starting with 0) by 1, indicating for any point **p** such that **S[i] <= p <= S[i+1]**,  if **S[i+1]** is an ending point, otherwise **S[i] <= p < S[i+1]** will be contained by that many of segments. If we find an ending point **S[i]**, we know for sure we had the opening point of that segment before, so it's time to close the segment, so we decrement the **counter** by 1, now indicating for any point **p** such that **S[i] < p <= S[i+1]**,  if **S[i+1]** is an ending point, otherwise **S[i] < p < S[i+1]** will be contained by that many of segments.
+Consider the set of all opening and ending points of the segments sorted in increasing order. Iterating over the set from the beginning, only two case can arise.
```
Comment (rebornplusplus) [inline 1089/en.md:13]:
```suggestion
Consider the set of all opening and ending points of the segments sorted in increasing order. Iterating over the set, there may be two scenarios: 
```

File: 1089/en.md [hunk 5]
```diff
@@ -10,15 +10,19 @@ First let's simplify the problem a bit. Suppose there is no such segment like: *
...
+Consider the set of all opening and ending points of the segments sorted in increasing order. Iterating over the set from the beginning, only two case can arise. 
 
-We can sort our queries so that we can calculate answers for them in increasing order as we iterate over the set **S**.
+1. An opening point **p** indicating the segment (of that opening point **p**) has started running. By means of saying that any point that comes after it (or equal to that) and not after the ending of it will be contained in that segment.
```
Comment (rebornplusplus) [inline 1089/en.md:15]:
```suggestion
1. An opening point `A` of segment `AB` is found. `AB` has started. Any point that comes after `A` (or equal to `A`) until we reach the endpoint `B` will be on that segment.
```

File: 1089/en.md [hunk 6]
```diff
@@ -10,15 +10,19 @@ First let's simplify the problem a bit. Suppose there is no such segment like: *
...
+Consider the set of all opening and ending points of the segments sorted in increasing order. Iterating over the set from the beginning, only two case can arise. 
 
-We can sort our queries so that we can calculate answers for them in increasing order as we iterate over the set **S**.
+1. An opening point **p** indicating the segment (of that opening point **p**) has started running. By means of saying that any point that comes after it (or equal to that) and not after the ending of it will be contained in that segment. 
 
-That was the general approach. Now, solving the original problems is all about handling the exception case where we may have the same point for many openings and endings. See the code and documentation for understading how to handle duplicate points.
+2. An ending point **p** indicating an ongoing segment has just been closed now and we can't take any point after that into the account of that segment (whose ending is **p**).
```
Comment (rebornplusplus) [inline 1089/en.md:17]:
```suggestion
2. An ending point `B` of segment `AB` is found. Ongoing segment `AB` is closed. We can't take any point after that into account. 
```

**Review by rebornplusplus — CHANGES_REQUESTED** [review]
Have some suggestions and the unresolved comment from the last pass.

**Comment by Talha-Taki002** [comment]
All Done! Thanks, bro

**Review by rebornplusplus — APPROVED** [review]
Looks good, thanks.

**Comment by rebornplusplus** [comment]
Resolves #449
